pkg/controller/scansettingbinding/scansettingbinding_controller.go

```diff
@@ -211,6 +214,52 @@ func (r *ReconcileScanSettingBinding) Reconcile(ctx context.Context, request rec
 		}
 	}
 
+	scanSetting := &compliancev1alpha1.ScanSetting{}
+	if instance.SettingsRef != nil {
+		ssb := instance.DeepCopy()
+		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: instance.SettingsRef.Name, Namespace: request.Namespace}, scanSetting)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("Failed to get ScanSetting: %s", err)
+		}
+		if scanSetting.ComplianceSuiteSettings.Suspend {
+			log.Info("Suspending ScanSettingBinding", "ScanSettingBinding", ssb.Name)
+			ssb.Status.SetConditionSuspended()
+			ssb.Status.Phase = compliancev1alpha1.ScanSettingBindingPhaseSuspended
+			err := r.Client.Status().Update(context.TODO(), ssb)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("Failed to update ScanSettingBinding status: %s", err)
+			}
+			cs := compliancev1alpha1.ComplianceSuite{}
+			err = r.Client.Get(context.TODO(), types.NamespacedName{Name: instance.Name, Namespace: request.Namespace}, &cs)
+			// If the ComplianceSuite doesn't exist, then we should
+			// return and not requeue the request because the
+			// ScanSettingBinding is being created from a suspended
+			// state and we don't want to create the
+			// ComplianceSuite or kick off any scans, yet. If or
+			// when the ScanSetting is activated, we will proceed
+			// like normal and create the ComplianceSuite and its
+			// ComplianceScans.
+			if errors.IsNotFound(err) {
+				return reconcile.Result{}, nil
+			}
+
+		} else if !scanSetting.ComplianceSuiteSettings.Suspend {
+			log.Info("Resuming ScanSettingBinding", "ScanSettingBinding", ssb.Name)
+			ssb.Status.SetConditionReady()
+			ssb.Status.Phase = compliancev1alpha1.ScanSettingBindingPhaseReady
+			err := r.Client.Status().Update(context.TODO(), ssb)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("Failed to update ScanSettingBinding status: %s", err)
+			}
+		}
+	}
+
+	instance = &compliancev1alpha1.ScanSettingBinding{}
```

Comment:
so maybe here we first check if the status needs to be updated by comparing the phase with `suite.spec.Suspend`, then if they are not matching, we update and then reconcile, if they do match we do nothing, and then in the next step, if the phase is matching what we wanted, we next check the phase, based on the phase, if the phase is in pausing, we check if there is existing suite exit, if they do we will need to run  `if suiteNeedsUpdate(&suite, &found)` and if they don't exist we do not create suit and reconcile.

Reply:
so if we do that i think we don't have to fetch the ssb instance one more time here

Reply:
Nice catch - fixed.